v2/contribute/decisions/accountlinking/0005-allowing-email-verification-in-post-login-account-linking.mdx

```diff
@@ -0,0 +1,66 @@
+- User is currently logged in to there account with email A (which has an associated primaryUserId)
+- User wants to link account with email B via email password flow.
+- Post login account linking API is called with email A's session and email B (and password )as new account input
+- Now because email B for email password recipe can only be verified once the recipe user is created, currently it's in unverified state as no recipe user even exists with email B. Also, we are not allowing recipe user account creation in this scenario.
+- Based on current flow, it would be impossible to allow linking of B to A unless email password account for B is already created and email B is verified.
+- To counter problem statement, we should
+    - allow creating recipe user even if it is unverified (and there doesn't exists any other account with email B that has a primaryUserId)
+    - trigger email verification flow for the recipe user with email B. The recipe user may be in existance prior to this API is called or it may be a new account to be created. In both scenarios, we would like to trigger a email verfication flow
+
+## Considered Options
+
+* allow account creation if necessary and trigger email verification flow
+
+
+## Decision Outcome
+
+Chosen option: allow account creation if necessary and trigger email verification flow, because
+
+* We also are allowing email verification flow to happen in the post login account linking API when the “new” account to link used to exist from before. The reason this is OK, is that in order for the new account to be linked in this API, the user has to login with that new account anyway whilst being logged in to their existing account. In order to login to the new account, they have to either have access to the social account, or the email ID, or know the password of that account.
+
+* Earlier we just threw an error saying not verified, which did not serve any purpose. We did this cause, we thought of some attack vectors example:
+    - Secenario 1:
+        - Attacker creates an account with email A, which belongs to them and a primaryUserId gets created
+        - Attacker tries to create an account with email B which doesn't belong to them will send a verification mail to actual user.
+        - Actual user clicks the verification link and the email password account gets linked to the attacker's account.
+        - If later the actual user signs up with email B via some other flow, those accounts will be linked to the attacker's account
+    - Scenario 2:
+        - There already exists an account with email B which is not associated with a primaryUserId because the actual user has not verified the account
+        - Attacker creates an account with email A, which belongs to them and a primaryUserId gets created
+        - Now attacker initiate post login account linking flow where the session exists with email A and account to be linked is B (which already exists but without primaryUserId)
```

Comment:
> Now attacker initiate post login account linking flow where the session exists with email A and account to be linked is B (which already exists but without primaryUserId)

How? Cause for this to happen, the attacker will need to know B's password - which they won't know right? So this scenario 2 is totally wrong?